crates/red_knot_test/src/parser.rs

```diff
@@ -303,6 +332,8 @@ impl<'s> Parser<'s> {
             // CODE_RE can't match without matches for 'lang' and 'code'.
             lang: captures.name("lang").unwrap().into(),
             code: captures.name("code").unwrap().into(),
+
+            starting_line_number: self.line_index.line_index(self.md_offset),
```

Comment:
I suggest just storing the embedded file and compute the line number when needed. It has the advantage that the test framework doesn't need to compute the `LineIndex` if there are no errors. Ccomputing the `LineIndex` isn't expensive, but it can add up. Quering the line index is `O(log(n))`, so that's not entirely free as well.